lib/vdsm/storage/storageServer.py

```diff
@@ -387,9 +387,21 @@ def _get_gluster_volinfo(self):
             volinfo = superVdsmProxy.glusterVolumeInfo(self._volname,
                                                        self._volfileserver)
             return volinfo[self._volname]
-        except ge.GlusterCmdExecFailedException as e:
-            log.warning("Failed to get volume info: %s", e)
-            return {}
+        except ge.GlusterException as e:
+            # The remote host may be down.
+            # If we are running on a hyperconverged system the gluster client
+            # can use one of the other gluster servers.
+            log.info("Failed to get volume info from remote server %s: %s",
+                     self._volfileserver, e)
+            try:
+                log.debug("Trying to get volume info from backup servers: %s",
+                          self._options)
```

Comment:
Logging the mount options in debug level makes sense, but we don't need to log inside the try block. We try to keep try blocks small as possible to avoid incorrect error handling.

Reply:
Suggested change implemented.